Software/ESPHome/office-blinds.yaml

```diff
@@ -0,0 +1,263 @@
+      - stepper.stop: motor
+      - lambda: id(sensored_home_pos) = id(encoder)->get_state();
+      - cover.template.publish:
+          id: blinds
+          state: OPEN
+      - light.turn_on:
+          id: led1
+          transition_length: 0s
+      - delay: 250ms
+      - light.turn_off:
+          id: led1
+          transition_length: 1s
+
+button:
+  - platform: restart
+    name: Restart
+
+  - platform: template
+    name: Home
+    id: home
+    on_press:
+      - stepper.set_target:
+          id: motor
+          target: -${encoder_closed_pos}
+
+  - platform: template
+    name: Stop
+    on_press:
+      - stepper.stop: motor
+
+binary_sensor:
+ - platform: gpio
+   name: Power Good
+   pin:
+     number: 15
+     mode: INPUT
+     inverted: true
+   device_class: power
+   filters:
+     - delayed_on: 10ms
+
+ - platform: gpio
+   name: Button 1
+   pin:
+     number: 35
+     mode: INPUT
+     inverted: true
+   id: btn1
+   filters:
+     - delayed_on: 10ms
+   on_press:
+     - cover.open: blinds
+
+ - platform: gpio
+   name: Button 2
+   id: btn2
+   pin:
+     number: 36
+     mode: INPUT
+     inverted: true
+   filters:
+     - delayed_on: 10ms
+   on_press:
+     - tmc2209.disable: motor
+
+ - platform: gpio
+   name: Button 3
+   id: btn3
+   pin:
+     number: 37
+     mode: INPUT
+     inverted: true
+   filters:
+     - delayed_on: 10ms
+   on_press:
+     - cover.close: blinds
+
+as5600:
+  slow_filter: 16x
+
+sensor:
+  - platform: as5600
+    id: encoder_status
+    update_interval: 60s
+    status:
+      name: Encoder status
+    
+  - platform: as5600
+    name: Encoder
+    id: encoder
+    update_interval: 0s # beware of the polling rate
+    filters:
+      - delta: 1 # throttle the high polling rate to only act on value changes
+      # compute absolute position from angle value
+      - lambda: |
+          const uint8_t curr_trirant = clamp<float>(floor(x / 1365.0), 0, 2);
+          const uint8_t prev_trirant = clamp<float>(floor(id(encoder_tracking_)[0] / 1365.0), 0, 2);
+          const int16_t encoder_diff = (4095 - x + id(encoder_tracking_)[0]);
+
+          if ((curr_trirant == 2 && prev_trirant == 0)) {
+            id(encoder_tracking_)[1] -= encoder_diff; // crossed zero clockwise
+          } else if ((curr_trirant == 0 && prev_trirant == 2)) {
+            id(encoder_tracking_)[1] += encoder_diff; // crossed zero counterclockwise
+          } else {
+            id(encoder_tracking_)[1] -= (id(encoder_tracking_)[0] - x);
+          }
+          id(encoder_tracking_)[0] = x;
+          return id(encoder_tracking_)[1];
+      - multiply: -1.0
+    accuracy_decimals: 0
+    state_class: measurement
+
+  - platform: adc
+    pin: 4
+    name: VBUS Voltage
+    update_interval: 10s
+    attenuation: 12dB
+    filters:
+      - multiply: 8.47742
+
+
+
+cover:
+  - platform: template
+    id: blinds
+    name: Blinds
+    has_position: true
+    assumed_state: false
+    internal: true
```

Comment:
Any reason why this is set to internal?

Reply:
Yep, this should definitely not be internal only - woops!